File: runme/mcgen/local/datan/datsrc/sqstud.c
#include "datsrc.h"

double LIBFUNC sqstud_(double const P_T *p, integer const P_T *n)
{
    /* Table of constant values */

    integer c__0 = 0;
    double c_d1em6 = 1e-6;

    /* System generated locals */
    double ret_val;

    /* Local variables */
    double xzero, x0, x1;

/* boundary of range */
    if (*p >= 1.) {
        ret_val = 1e10;
    }
    if (*p <= 0.) {
        ret_val = -1e10;
    }
/* normal range */
    if (*p < 1. && *p > 0.) {
        x0 = 0.;
        x1 = *p;
        auxzbr_(&x0, &x1, szstud_, p, n, &c__0);
        auxzfn_(&x0, &x1, &xzero, szstud_, p, n, &c__0, &c_d1em6);
        ret_val = xzero;
    }
    return ret_val;
} /* sqstud_ */

/* ----------------------------------------------------------------- */
double LIBFUNC szstud_(double const P_T *x, double const P_T *p,
                      integer const P_T *n, integer const P_T *ndum)
{
    /* System generated locals */
    double ret_val;

    /* Local variables */

/* returns P minus cumulative Student's distribution of (X,N) */
    ret_val = *p - scstud_(x, n);
    return ret_val;
} /* szstud_ */

File: runme/mcgen/local/datan/datsrc/sqstud.f
      DOUBLE PRECISION FUNCTION SQSTUD(P,N)
      IMPLICIT DOUBLE PRECISION(A-H,O-Z)
      PARAMETER(BIG=1D10,EPSILN=1D-6,ONE=1.D0,ZERO=0.D0,HALF=.5D0)
      EXTERNAL SZSTUD
C boundary of range
      IF(P.GE.ONE) SQSTUD=BIG
      IF(P.LE.ZERO) SQSTUD=-BIG
C normal range
      IF(P.LT.ONE .AND. P.GT.ZERO) THEN
        X0=ZERO
        X1=P
        CALL AUXZBR(X0,X1,SZSTUD,P,N,0)
        CALL AUXZFN(X0,X1,XZERO,SZSTUD,P,N,0,EPSILN)
        SQSTUD=XZERO
      END IF
      END
C-----------------------------------------------------------------
      DOUBLE PRECISION FUNCTION SZSTUD(X,P,N,NDUM)
C returns P minus cumulative Student's distribution of (X,N)
      IMPLICIT DOUBLE PRECISION(A-H,O-Z)
      SZSTUD=P-SCSTUD(X,N)
      END
